clang/test/dpct/kernel-function-typecast.cu

```diff
@@ -0,0 +1,13 @@
+// RUN: dpct -out-root %T/kernel-function-typecast %s --cuda-include-path="%cuda-path/include"
+// RUN: FileCheck --match-full-lines --input-file %T/kernel-function-typecast/kernel-function-typecast.dp.cpp %s
+// RUN: %if build_lit %{icpx -c -fsycl %T/kernel-function-typecast/kernel-function-typecast.dp.cpp -o %T/kernel-function-typecast/kernel-function-typecast.dp.o %}
+
+#include <cstdint>
+#include <cuda.h>
+
+typedef uint64_t u64;
+u64 foo(CUfunction cuFunc, CUmodule cuMod) {
+  cuModuleGetFunction(&cuFunc, cuMod, "kfoo");
+  u64 function = (u64)cuFunc;
```

Comment:
Is the typecast necessary?
```suggestion
  u64 function = cuFunc;
```

Reply:
The test is to check explicit typecasting. Now we have two line for both implicit and explicit casting.